src/Tinfoil/Key.hs

```diff
@@ -0,0 +1,18 @@
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE OverloadedStrings #-}
+module Tinfoil.Key(
+    genSymmetricKey256
+) where
+
+import           P
+
+import           System.IO (IO)
+
+import           Tinfoil.Data.Key
+import           Tinfoil.Data.Random
+import           Tinfoil.Random
+
+-- | Generate a 256-bit symmetric cryptographic key.
+genSymmetricKey256 :: IO SymmetricKey
+genSymmetricKey256 = fmap (SymmetricKey . unEntropy) $ entropy 32
```

Comment:
Is there a reason to prefer it? Thought it would compile to the same code, at least with `IO`.